app/uk/gov/hmrc/thirdpartyorchestrator/domain/utils/EnumJson.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.thirdpartyorchestrator.domain.utils
+
+import play.api.libs.json._
+
+class InvalidEnumException(className: String, input: String)
+    extends RuntimeException(s"Enumeration expected of type: '$className', but it does not contain '$input'")
+
+object EnumJson {
+
+  def enumReads[E <: Enumeration](enumValue: E): Reads[E#Value] = new Reads[E#Value] {
```

Comment:
Would this be simpler with a sealed trait instead of Enumeration (?)

Reply:
Removed